flow2ml/Data_Augumentation.py

```diff
@@ -435,3 +435,59 @@ def applyClose(self,classPath):
             cv2.imwrite(classPath+"/ClosedImages/Closed"+image, Closed)
           except Exception as e:
             print(f"Closing operation failed due to {e}")
+
+  def applyThreshold(self,classPath):
+    ''' 
+      Applies thresholding augmentation to all the images in the given folder.
+      Args : 
+        classPath : (string) directory containing images for a particular class.
+    '''
+
+    try:
+      os.mkdir(classPath+"/ThresholdedImages")    
+    except:
+      raise Exception("Unable to create directory for thresholded images.")
+    for image in list(os.listdir(classPath)):
+      
+      # Read image
+      img = cv2.imread(classPath+"/"+image)
+      
+      if self.operations['threshold'] not in ['simple', 'adaptive', 'OTSU']:
+        raise Exception("Invalid threshold operation.")
+      else:
+        
+        if self.operations['threshold'] == 'simple':
+          if img is not None:
+            try:
+              #convert the image from BGR to GRAYSCALE
+              gray=cv2.cvtColor(img,cv2.COLOR_BGR2GRAY)
+              # applies simple thresholding augmentation to the image.
+              _,s_threshed=cv2.threshold(gray,120,255,cv2.THRESH_BINARY)  
+              # saving the image by
+              plt.imsave(classPath+"/ThresholdedImages/s_threshed"+image,s_threshed)
+            except Exception as e:
+              print(f"simple thresholding operation failed due to {e}")
+
+        elif self.operations['threshold'] == 'adaptive':
+          if img is not None:
+            try:
+              #convert the image from BGR to GRAYSCALE
+              gray=cv2.cvtColor(img,cv2.COLOR_BGR2GRAY)
+              # applies adaptive thresholding augmentation to the image.
+              a_threshed=cv2.adaptiveThreshold(gray,255,cv2.ADAPTIVE_THRESH_MEAN_C,cv2.THRESH_BINARY,199,5)  
+              # saving the image by
+              plt.imsave(classPath+"/ThresholdedImages/a_threshed"+image,a_threshed)
+            except Exception as e:
+              print(f"adaptive thresholding operation failed due to {e}")
+
+        elif self.operations['threshold'] == 'OTSU':
+          if img is not None:
+            try:
+              #convert the image from BGR to GRAYSCALE
+              gray=cv2.cvtColor(img,cv2.COLOR_BGR2GRAY)
+              # applies OTSU thresholding augmentation to the image.
+              _,o_threshed=cv2.threshold(gray,120,255,cv2.THRESH_BINARY+cv2.THRESH_OTSU) 
```

Comment:
How are you taking these threshold values?

Reply:
> How are you taking these threshold values? @ainy-123

In this method, threshold values are  not required to set. I took it just as random value.
 whatever threshold value we put, it doesn't matter actually.it will adjust the right threshold value.

Reply:
@rubyruins I meant not for adaptive one, I am asking for other techniques

Reply:
> @ainy-123 I meant not for adaptive one, I am asking for other techniques

I took the threshold value near to the half of range~(255/2).
Again threshold value depends on the user. But ,I took here this way, just as an example.

Reply:
Yes @ainy-123 
Please take inputs from user for those threshold values.

Reply:
> Yes @ainy-123
> Please take inputs from user for those threshold values.

changes done. Please have a look.